dev-libs/mini-xml/mini-xml-2.12.ebuild

```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit autotools
+
+MY_PN="mxml"
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="A small XML parsing library that you can use to read XML data files or strings"
+HOMEPAGE="https://github.com/michaelrsweet/mxml
+	https://www.msweet.org/mxml/"
+SRC_URI="https://github.com/michaelrsweet/${MY_PN}/archive/v${PV}.tar.gz -> ${MY_P}.tar.gz"
+
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+LICENSE="Mini-XML"
+SLOT="0"
+IUSE="static-libs threads"
+
+BDEPEND="virtual/pkgconfig"
+
+S="${WORKDIR}/${MY_P}"
+
+src_prepare() {
+	# Respect users CFLAGS
+	sed -e 's/-Os -g//' -i configure.ac || die
+
+	# Don't run always tests
+	# Enable verbose compiling
+	sed -e '/ALLTARGETS/s/testmxml//g' -e '/.SILENT:/d' -i Makefile.in || die
+
+	# Build only static-libs, when requested by user, also build docs without static-libs in that case
+	if ! use static-libs; then
+		local mysedopts=(
+			-e '/^install:/s/install-libmxml.a//g'
+			-e '/^mxml.xml:/s/-static//g'
```

Comment:
Does this mean that with USE=static-libs, installed executables are linked statically? We really do want to use dynamic linking *always*.

Reply:
No, all executables are still dynamic. Let me try to explain.

The Makefile ist just pretty shitty. It always builds static and dynamic lib.
When I disable compilation of static lib, aka USE=-static-libs, I have to patch arround:
1) Tests are always run and do compile with static libs for itself, so it's build, even, when I don't want that. So I patched this.
2) The creation of docs uses an selfbuild binary (it's never installed to the system), to create the docs. Per default, it's static. So again, it forces building the static lib, even, If a user don't want to. So patching this also around.